eclair-core/src/test/scala/fr/acinq/eclair/io/PeerConnectionSpec.scala

```diff
@@ -191,19 +193,19 @@ class PeerConnectionSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike wi
 
   test("sync if no whitelist is defined") { f =>
     import f._
-    val remoteInit = wire.Init(bin"10000000".bytes) // bob supports channel range queries
+    val remoteInit = wire.Init(Features(Set(ActivatedFeature(ChannelRangeQueries, Optional))))
     connect(remoteNodeId, switchboard, router, connection, transport, peerConnection, peer, remoteInit, expectSync = true)
   }
 
   test("sync if whitelist contains peer", Tag("sync-whitelist-bob")) { f =>
     import f._
-    val remoteInit = wire.Init(bin"0000001010000000".bytes) // bob supports channel range queries and variable length onion
+    val remoteInit = wire.Init(Features(Set(ActivatedFeature(ChannelRangeQueries, Optional), ActivatedFeature(VariableLengthOnion, Optional))))
     connect(remoteNodeId, switchboard, router, connection, transport, peerConnection, peer, remoteInit, expectSync = true)
   }
 
   test("don't sync if whitelist doesn't contain peer", Tag("sync-whitelist-random")) { f =>
     import f._
-    val remoteInit = wire.Init(bin"0000001010000000".bytes) // bob supports channel range queries
+    val remoteInit = wire.Init(Features(bin"0000001010000000")) // bob supports channel range queries
```

Comment:
I think it's not this test's responsibility to test the feature parsing from bits, better make it more readable:

```suggestion
    val remoteInit = wire.Init(Features(Set(ActivatedFeature(ChannelRangeQueries, Optional))))
```